buildsystem/Doxygen.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2014-2015 the openage authors. See copying.md for legal info.
+
+option(GENERATE_DOCS "Generate documentation from the source files." ON)
+
+set(DOXYFILE_TEMPLATE "${BUILDSYSTEM_DIR}/templates/Doxyfile.in")
+set(DOCS_ROOT_DIRECTORY "${CMAKE_BINARY_DIR}/docs")
+
+function(add_doxygen_target TARGET TARGET_SRC_DIR)
+	if(NOT DOXYGEN_FOUND)
+		return()
+	endif()
+
+	set(DOXYFILE "${CMAKE_BINARY_DIR}/${TARGET}_Doxyfile")
+	set(OUTPUT_DIRECTORY "${DOCS_ROOT_DIRECTORY}/${TARGET}")
```

Comment:
This implementation is almost same as before. The output folders will now become `bin/docs/libopenage` and `bin/docs/openage` instead of `bin/doc`. If this is ok, i can modify the doc readme file.